### PR TITLE
Channel to use ITelemetryProcessors from core.

### PR DESCRIPTION
**Pull request by cijothomas**

### CODE REVIEW

**Comment by cijothomas** [comment]
Modified the channel project to use ITelemetryProcessor from the core, and removed the Builders.
@upendras

File: Test/ServerTelemetryChannel.Test/Shared.Tests/SamplingTelemetryProcessorTest.cs [hunk 1]
```diff
@@ -63,83 +64,83 @@ public void TelemetryItemHasSamplingPercentageSet()
         [TestMethod]
         public void DependencyTelemetryIsSubjectToSampling()
         {
-            TelemetryTypeSupportsSampling((channel) => channel.Send(new DependencyTelemetry()));
+            TelemetryTypeSupportsSampling((telProcessors) => telProcessors.Process(new DependencyTelemetry()));
```
Comment (upendras) [inline Test/ServerTelemetryChannel.Test/Shared.Tests/SamplingTelemetryProcessorTest.cs]: nit - either processors or telemetryProcessors, please don't invent abbreviations

File: src/TelemetryChannels/ServerTelemetryChannel/Shared/TelemetryChannelBuilderExtensions.cs [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
-    /// </summary>
-    public static class TelemetryChannelBuilderExtensions
-    {
-        /// <summary>
-        /// Adds <see cref="SamplingTelemetryProcessor"/> to the <see cref="TelemetryChannelBuilder" />
-        /// </summary>
-        /// <param name="builder">Instance of <see cref="TelemetryChannelBuilder"/></param>
-        /// <param name="samplingPercentage">Sampling Percentage to configure.</param>
-        /// <return>Instance of <see cref="TelemetryChannelBuilder"/>.</return>
-        public static TelemetryChannelBuilder UseSampling(this TelemetryChannelBuilder builder, double samplingPercentage)
```
Comment (upendras) [inline src/TelemetryChannels/ServerTelemetryChannel/Shared/TelemetryChannelBuilderExtensions.cs]: this extension is still useful, lets not remove.

File: Test/ServerTelemetryChannel.Test/Shared.Tests/Helpers/StubTelemetryProcessor.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
     using System;
 
     using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
```
Comment (SergeyKanzhelev) [inline Test/ServerTelemetryChannel.Test/Shared.Tests/Helpers/StubTelemetryProcessor.cs]: why this change?

Comment (cijothomas) [inline Test/ServerTelemetryChannel.Test/Shared.Tests/Helpers/StubTelemetryProcessor.cs]: to refer to the ItelemetryProcessor in the Extensibility namespace.
